logprep/framework/pipeline.py

```diff
@@ -258,28 +286,23 @@ def _get_event(self) -> dict:
     def process_event(self, event: dict):
         """process all processors for one event"""
 
-        event_received = self._encoder.encode(event)
-        extra_outputs = []
+        results = []
+
         for processor in self._pipeline:
-            try:
-                if extra_data := processor.process(event):
-                    if self._output:
-                        self._store_extra_data(extra_data)
-                    extra_outputs.append(extra_data)
-            except ProcessingWarning as error:
-                self.logger.warning(str(error))
-            except ProcessingCriticalError as error:
-                self.logger.error(str(error))
+            result: ProcessorResult = processor.process(event)
+            results.append(result)
+            if result.data:
                 if self._output:
-                    self._store_failed_event(error, copy.deepcopy(event), event_received)
-                    event.clear()
+                    self._store_extra_data(result)
+            if ProcessingError in result:
+                event.clear()
             if not event:
                 break
-        return extra_outputs
+        return PipelineResult(results=results)
 
```

Comment:
Not possible because we need to interrupt the pipeline once the deleter clears the event. Alternatively we could pass an empty event through the pipeline by adding an if statement into the apply_rules_wrapper method, which checks if the event is cleared, to ensure early stopping of each processor. Downside is that this is evaluated for every processor and every event indepentently of it being cleared or not.

Reply:
I just added a solution that looks like this:
```python
    @Metric.measure_time()
    def process_event(self, event: dict):
        """process all processors for one event"""
        return PipelineResult(
            results=[processor.process(event) for processor in self._pipeline if event]
        )
```
Now the only issue, as @djkhl mentioned already, is that we execute this litte `if event` for every processor, instead of breaking out early. This should be still faster than checking it first inside the processor itself, especially because it is still happening inside the list comprehension. To make this work I had to clear the event though inside the `_apply_rules_wrapper` method inside the `processor.py`. Maybe this could be a solution which we just have to try out.